### PR TITLE
Small changes to yul object specification.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/yul.rst [hunk 1]
```diff
@@ -391,7 +391,7 @@ The following functions must be available:
 +---------------------------------------------+-----------------------------------------------------------------+
 | shru256(x:u256, y:u256) -> z:u256           | logical right shift of x by y                                   |
 +---------------------------------------------+-----------------------------------------------------------------+
-| sars256(x:s256, y:u256) -> z:u256           | arithmetic right shift of x by y                                |
+| saru256(x:u256, y:u256) -> z:u256           | arithmetic right shift of x by y                                |
```
Comment (axic) [inline docs/yul.rst]: Are you sure about these? I think we've changed them.

File: docs/yul.rst [hunk 2]
```diff
@@ -515,6 +515,16 @@ The following functions must be available:
...
 | keccak256(p:u256, s:u256) -> v:u256         | keccak(mem[p...(p+s)))                                          |
 +---------------------------------------------+-----------------------------------------------------------------+
+| *Object access*                             |                                                                 |
++---------------------------------------------+-----------------------------------------------------------------+
+| datasize(name:string) -> size:u256          | size of the data object in bytes, name has to be string literal |
++---------------------------------------------+-----------------------------------------------------------------+
+| dataoffset(name:string) -> offset:u256      | offset of the data object inside the data area in bytes,        |
+|                                             | name has to be string literal                                   |
++---------------------------------------------+-----------------------------------------------------------------+
+| datacopy(t:u256, o:u256, s:u256)            | copy s bytes from the data area starting at offset o bytes      |
```
Comment (axic) [inline docs/yul.rst]: t, o, s => dst, src, len

File: docs/yul.rst [hunk 3]
```diff
@@ -558,13 +575,16 @@ An example Yul Object is shown below:
     // Code consists of a single object. A single "code" node is the code of the object.
     // Every (other) named object or data section is serialized and
     // made accessible to the special built-in functions datacopy / dataoffset / datasize
-    object {
+    // Access to nested objects can be performed by joining the names using ``.``.
```
Comment (axic) [inline docs/yul.rst:578]: Needs an example for nested objects.

Comment (chriseth) [reply docs/yul.rst:578]: Added.

File: docs/yul.rst (hunk 1 above)
Comment (chriseth) [reply docs/yul.rst]: Argh!

Comment (chriseth) [reply docs/yul.rst]: Changed.

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
```
Warning, treated as error:
/root/project/docs/solidity-by-example.rst:689: ERROR: Unexpected indentation.
```

It seems there is an unrelated failure, probably some previous PR was broken.

**Pull request merged**